### ISSUE TITLE
LoggingTriggerHistoryPlugin and LoggingJobHistoryPlugin should use StructuredLogging

### ISSUE THREAD

**Issue by michaelmairegger** [quartznet/quartznet#2217]
Currently the log is written as string. For better logging those plugins should use structured loggings.

Additionally to that every call to Log.XXX should use structured logging.

**Comment by jafin** [comment]
I started to take a look at this, once structured logging is introduced, I think it makes less sense to provide ability to configure your own log template message, which the current plugin does.  As the order of the parameters, not the placeholder index takes effect.   I'll put up a PR for review, and see if this is acceptable.

**Comment by michaelmairegger** [comment]
https://github.com/quartznet/quartznet/blob/ee42e22b0d376018a584ca31a4b083a1b7a2d094/src/Quartz.Plugins/Plugin/History/LoggingJobHistoryPlugin.cs#L469

This line disables **CA2254: Template should be a static expression**.

@jafin any news on structured logging?

**Comment by michaelmairegger** [comment]
@lahma what do you think about that? I think using structured logging should be essential. For LoggingTriggerHistoryPlugin and LoggingJobHistoryPlugin structured logging is missing.
The only issue to introducte this for both classes is, that there is a xxxMessage with custom Template {0},...{1} which should not be used in structured logging. Is the possibility to modify the message template such important?

**Comment by lahma** [comment]
If you look at the amount of code these plugins have, it might not be that hard to write your own version with hard-coded message templates which cater to your needs.

**Comment by michaelmairegger** [comment]
no, but it would be nice to have it in core library. if desired I can provide a PR

**Comment by lahma** [comment]
Well I think it might make sense to have `Structured` prefixed ones as an alternative (without configuration), maybe have them to use some modern .NET targeting sprinkle - basically supported on modern platforms or similar. I'm open to ideas.

**Comment by michaelmairegger** [comment]
since LibLog will be removed in 4.0 it might be an ideal point to use structured logging by default. I saw that the main branch the major parts of the code use Microsoft.Extensions.Logging.ILogger.
If desired I can migrate `LoggingTriggerHistoryPlugin` and `LoggingJobHistoryPlugin` in main branch.

**Comment by lahma** [comment]
I guess the change would mean that we remove the templates or do you mean we still support templates too?

**Comment by michaelmairegger** [comment]
~Which template do you mean? can you point to a specific line of code?~

Ah, I think you mean the templates in LoggingTriggerHistoryPlugin. Hmm, yes, when using ILogger, the templates will not work the same way. The question is: Is there really a need for having a customizzable logging message?

**Comment by lahma** [comment]
> ~Which template do you mean? can you point to a specific line of code?~
> 
> Ah, I think you mean the templates in LoggingTriggerHistoryPlugin. Hmm, yes, when using ILogger, the templates will not work the same way. The question is: Is there really a need for having a customizzable logging message?

This is the thing, if people want semantic logging and we would be forced to dictate the format how much do we win anymore. Should the logging be core part of normal execution pipeline and then checked against Verbose logging or similar.
